packages/block-editor/src/components/block-settings-menu/block-edit-visually-button.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * WordPress dependencies
+ */
+import { rawHandler, getBlockContent } from '@wordpress/blocks';
+import { ToolbarButton, ToolbarGroup } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { useCallback } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { store as blockEditorStore } from '../../store';
+
+export default function BlockEditVisuallyButton( { clientIds, ...props } ) {
+	const { block, shouldRender } = useSelect(
+		( select ) => {
+			const firstBlockClientId = clientIds[ 0 ];
+			const { isBlockMultiSelected, getBlockMode, getBlock } = select(
+				blockEditorStore
+			);
+			const isSingleSelected = ! isBlockMultiSelected(
+				firstBlockClientId
+			);
```

Comment:
Current unit test failure is caused by the recent update to `wp-prettier` in `trunk`. (See #40542.)

Here's what the reformated code should look like:
```suggestion
			const { isBlockMultiSelected, getBlockMode, getBlock } =
				select( blockEditorStore );
			const isSingleSelected =
				! isBlockMultiSelected( firstBlockClientId );
```

Reply:
Since that's the last required change and it's just code formatting, I went ahead and clicked "commit"